### PR TITLE
Explain semantics of subqueries

### PR DESCRIPTION
**Pull request by hedengran**

### CODE REVIEW

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/CallSubqueryTest.scala [hunk 1]
```diff
@@ -73,6 +81,43 @@ class CallSubqueryTest extends DocumentingTest {
       graphViz()
     }
 
+    section("Semantics, call-semantics") {
```
Comment (blouerat) [inline cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/CallSubqueryTest.scala:84]: Didn't you mean to have the comma outside the string?
```suggestion
    section("Semantics", "call-semantics") {
```

Comment (hedengran) [reply cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/CallSubqueryTest.scala:84]: Yes. Thank you.

**Review by blouerat — APPROVED** [review]

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/CallSubqueryTest.scala [hunk 2]
```diff
@@ -73,6 +81,43 @@ class CallSubqueryTest extends DocumentingTest {
...
+      p(
+        """A `CALL` clause is executed once for each incoming row.
+          #For the below query, the `CALL` clause will execute three times.
```
Comment (AlexicaWright) [inline cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/CallSubqueryTest.scala]:
```suggestion
          #For the below query, the `CALL` clause executes three times.
```

**Review by AlexicaWright — COMMENTED** [review]
Just one editorial comment, but I haven't been able to build locally.

**Comment by martin-neotech** [comment]
@hedengran please rebase

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/CallSubqueryTest.scala [hunk 3]
```diff
@@ -73,6 +81,43 @@ class CallSubqueryTest extends DocumentingTest {
...
+
+      p(
+        """Each execution of a `CALL` clause can observe changes from previous executions.
+          #""".stripMargin('#'))
+
+      query("""UNWIND [0, 1, 2] AS x
+              #CALL {
+              #  MATCH (n:Counter)
+              #  SET n.count = n.count + 1
+              #  RETURN n.count as innerCount
```
Comment (martin-neotech) [inline cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/CallSubqueryTest.scala]:
```suggestion
              #  RETURN n.count AS innerCount
```

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/CallSubqueryTest.scala [hunk 4]
```diff
@@ -73,6 +81,43 @@ class CallSubqueryTest extends DocumentingTest {
...
+              #RETURN innerReturn""".stripMargin('#'),
+        ResultAssertions(r => r.toSet should equal(Set(
+          Map("innerReturn" -> "hello"),
+          Map("innerReturn" -> "hello"),
+          Map("innerReturn" -> "hello"),
+        )))) { resultTable() }
+
+      p(
+        """Each execution of a `CALL` clause can observe changes from previous executions.
+          #""".stripMargin('#'))
```
Comment (martin-neotech) [inline cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/CallSubqueryTest.scala]:
```suggestion
      p("Each execution of a `CALL` clause can observe changes from previous executions.")
```

**Review by martin-neotech — CHANGES_REQUESTED** [review]
Looks good, just some minor changes needed. Thanks

**Review by martin-neotech — APPROVED** [review]

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/CallSubqueryTest.scala [hunk 5]
```diff
@@ -73,6 +81,41 @@ class CallSubqueryTest extends DocumentingTest {
       graphViz()
     }
 
+    section("Semantics, call-semantics") {
+      p(
+        """A `CALL` clause is executed once for each incoming row.
+          #For the below query, the `CALL` clause will executes three times.
```
Comment (AlexicaWright) [inline cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/CallSubqueryTest.scala]:
```suggestion
          #For the query below, the `CALL` clause executes three times.
```

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/CallSubqueryTest.scala [hunk 6]
```diff
@@ -94,6 +137,41 @@ class CallSubqueryTest extends DocumentingTest {
...
+      caution{
+        p(
+          """The order in which subqueries are executed is not defined.
+            #If a query result depends on the order of execution of subqueries, an `ORDER BY` clause should precede the `CALL` clause.
+            #""".stripMargin('#'))
+      }
+
+      p(
+        """The below query creates a linked list of all `:Person` nodes in order of ascending age.
+          #The `CALL` clause is relying on the incoming row ordering to ensure that a correctly linked list is created, so the incoming rows must be ordered with a preceding `ORDER BY`.
```
Comment (AlexicaWright) [inline cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/CallSubqueryTest.scala]:
```suggestion
          #The `CALL` clause is relying on the incoming row ordering to ensure that a correctly linked list is created, thus the incoming rows must be ordered with a preceding `ORDER BY`.
```

**Review by AlexicaWright — COMMENTED** [review]
Just a few editorial comments.

**Pull request merged**